src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -59,15 +59,9 @@ private BoundStatement MakeLoweredForm(BoundPatternSwitchStatement node)
                 var expression = _localRewriter.VisitExpression(node.Expression);
                 var result = ArrayBuilder<BoundStatement>.GetInstance();
 
-                // if the expression is "too complex", we copy it to a temp.
```

Comment:
Lowering starts by lowering to a decision tree, which starts by calling `CreateEmptyDecisionTree`, which contains code equivalent to this deleted code. The existing generated code always contained two temps instead of one.

``` c#
        /// <summary>
        /// Create a fresh decision tree for the given input expression of the given type.
        /// </summary>
        protected DecisionTree CreateEmptyDecisionTree(BoundExpression expression)
        {
            var type = expression.Type;

            LocalSymbol temp = null;
            if (expression.ConstantValue == null)
            {
                // Unless it is a constant, the decision tree acts on a copy of the input expression.
                // We create a temp to represent that copy. Lowering will assign into this temp.
                temp = PatternMatchingTemp(type);
                expression = new BoundLocal(expression.Syntax, temp, null, type);
            }
```